app/controllers/dashboards_controller.rb

```diff
@@ -1,4 +1,30 @@
 # frozen_string_literal: true
 class DashboardsController < ApplicationController
-  def index; end
+  include Flashable
+
+  def index
+    @uploads = Upload.last_uploads
+  end
+
+  def build
+    @version = InstitutionBuilder.run(current_user)
```

Comment:
We've tweaked timeouts for GIDS already at rev proxy and load balancer layers so unless this is vastly slower than the current production's export/data push operations I think it will succeed. If not, it will result in a "Request timed out" error (probably from nginx) but the operation will continue running in the GIDS app.